### ISSUE TITLE
Staging Review - Accessibility Feedback - Converted Segmented progress bar / Step Indicator to USWDS v3

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#59934]
## VFS actions
- [ ] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] Close the ticket when all feedback has been addressed.

## Thoughts/questions

-

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Must:** Right now, the "X of Y Heading Text" line is hard-coded as an `h4`, but the actual heading level should depend on the context in which the progress bar is being used. (USWDS has a [note about heading levels](https://designsystem.digital.gov/components/step-indicator/#accessibility-step-indicator) in their guidance too.) I think it makes sense for there to be a `level` property for setting heading level of the component, similar to what exists on the `va-accordion-item` component. 
- **Consider:** Better yet, thinking about what might be coming down the road --- the Forms team is working on a [simplified form header](https://www.sketch.com/s/9f215f8f-ba86-4b99-813f-d0ca3c03ca18/v/a7Oqwq/p/F1BE1800-65B1-40C5-B19A-E200BE1C567A/canvas?posX=-1817.7776566407592&posY=-590.7187485389238&zoom=0.5706323981285095) in which the step indicator would be a `div` with an `aria-describedby` attribute rather than a heading on its own. If possible, the component should support both a heading plus heading level and a non-semantic implementation like what the Forms team has in mind.
- **Consider:** The v3 component gets rid of the word "Step" from the visible text, but still has "Step" announced as screen reader only text for the "Step X of Y Heading Text" heading. I don't have any specific problems with that.
   
   But thinking about the Forms team's simplified header again, they're exploring phrasing each segment of the process as a "Chapter" rather than as a "Step." Individual segments of a form might have several tasks, and calling a collection of several tasks a single step could be confusing. If you want to support implementations like what the Forms team is thinking, it would be good to include a component property that allows for customizing that screen reader text. I think "Step" still makes sense as the default but a prop that overrides the default would provide some flexibility.
- **Should:** The `label` property adds an `aria-label` to the `usa-step-indicator` div that wraps the whole component. This is consistent with the USWDS' recommendation to put an `aria-label="progress"` on the component. But I think that guidance is wrong! 
   
   In general, screen readers don't announce `aria-label` when it appears on a `div` and `aria-label` isn't really intended for non-semantic elements. (I believe [this table on aria-label support from Steve Faulkner](https://html5accessibility.com/stuff/2020/11/07/not-so-short-note-on-aria-label-usage-big-table-edition/) is still accurate.) It doesn't really make anything worse but it doesn't serve any real purpose --- and since you also have a `labels` property that does something else for the component, if you can get rid of `label` you'll probably reduce developer confusion!
   
   With regard to the intent of that `aria-label`, the USWDS documentation says: "Placing this aria-label on the element with the class usa-step-indicator helps provide important context to screen readers." My gut reaction is that the extra context probably isn't needed, but if the USWDS team has research backing this up I'll gladly defer to them. In that case, it's better to use `sr-only` text inside the `div` and before the `ol`.
- **Must:** The USWDS documentation recommends using `aria-current` on the current item in the list. I'm not seeing that in this component.
   - Note that USWDS recommends using `aria-current="true"`, which is fine. But the attribute also supports `aria-current="step"`, which I believe is well-supported and is a better fit for this use case.
- **Must:** The USWDS documentation recommends using `aria-hidden="true"` on the `ol` element wrapping the segments when no step labels are used. I'm not seeing that in this component. I think most screen readers will probably handle this fine without `aria-hidden`, but it's possible some may announce something like "list with X items" and then not announce any items, which could be confusing. The `aria-hidden` prevents that from being an issue.

**Comment by powellkerry** [comment]
PR: https://github.com/department-of-veterans-affairs/component-library/pull/805
